This patch is only needed for old kernels that do not have SOCK_CLOEXEC flag - used in tomato
--- a/src/fdevent.c
+++ b/src/fdevent.c
@@ -16,12 +16,22 @@
 
 #ifndef _WIN32
 
+#ifndef LINUX_VERSION_CODE
+#include <linux/version.h>
+#endif
+
+/* SOCK_CLOEXEC & SOCK_NONBLOCK since Linux 2.6.27 */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,27)
 #ifdef SOCK_CLOEXEC
 static int use_sock_cloexec;
 #endif
 #ifdef SOCK_NONBLOCK
 static int use_sock_nonblock;
 #endif
+#else
+#undef SOCK_CLOEXEC
+#undef SOCK_NONBLOCK
+#endif /* SOCK_CLOEXEC & SOCK_NONBLOCK since Linux 2.6.27 */
 
 void fdevent_socket_nb_cloexec_init (void)
 {
